rsocket-core/src/main/java/io/rsocket/core/StateUtils.java

```diff
@@ -13,27 +13,36 @@
   /** Bit Flag that indicates Requester Producer has been subscribed once */
   static final long SUBSCRIBED_FLAG =
       0b000000000000000000000000000000001_0000000000000000000000000000000L;
+  /** Bit Flag that indicates that the first payload in RequestChannel scenario is received */
+  static final long FIRST_PAYLOAD_RECEIVED_FLAG =
+      0b000000000000000000000000000000010_0000000000000000000000000000000L;
+  /**
+   * Bit Flag that indicates that the logical stream is prepared for the first initial frame sending
+   * (applicable for requester only)
+   */
+  static final long PREPARED_FLAG =
+      0b000000000000000000000000000000100_0000000000000000000000000000000L;
```

Comment:
Maybe `READY_TO_SEND_FIRST_FRAME_FLAG` and also if the Javadoc could reflect that this is set either when a lease permit is acquired, or otherwise when `request` is received, if leasing is not enabled.

Reply:
applied suggestion